BDArmory/Control/MissileFire.cs

```diff
@@ -2933,6 +2949,17 @@ public void UpdateList()
                             if (weap.Current == null) continue;
                             if (weap.Current.GetShortName() == weaponName)
                             {
+                                if (weapon.Current.GetWeaponClass() == WeaponClasses.Missile || weapon.Current.GetWeaponClass() == WeaponClasses.Bomb || weapon.Current.GetWeaponClass() == WeaponClasses.SLW)
+                                {
+                                    float range = weapon.Current.GetPart().FindModuleImplementing<MissileBase>().engageRangeMax;
+
+                                    if (rangeList.TryGetValue(weaponName, out var registeredRanges))
+                                    {
+                                        if (registeredRanges.Contains(range))
+                                            alreadyAdded = true;
+                                    }
+                                }
+                                else
```

Comment:
The main reason for the initial shortname method was working around how UpdateList worked; having a sorted list to compare to is slightly less performant, but updateList isn't getting called that often, and removes the potential of custom short names breaking things and associated side effects (missile range appearing in selected Weapon X popups, the marquee, etc)

Reply:
Why are you using a `SortedList` for this instead of a `Dictionary`? The ordering doesn't appear to be important.

Reply:
Good point. Changed to Dictionary